central/config/datastore/datastore.go

```
@@ -225,3 +239,84 @@ func clusterRetentionConfigsEqual(c1 *storage.DecommissionedClusterRetentionConf
 	return c1.GetRetentionDurationDays() == c2.GetRetentionDurationDays() &&
 		reflect.DeepEqual(c1.GetIgnoreClusterLabels(), c2.GetIgnoreClusterLabels())
 }
+
+func (d *datastoreImpl) GetPlatformComponentConfig(ctx context.Context) (*storage.PlatformComponentConfig, bool, error) {
+	if ok, err := administrationSAC.ReadAllowed(ctx); err != nil {
+		return nil, false, err
+	} else if !ok {
+		return nil, false, nil
+	}
+
+	config, found, err := d.store.Get(ctx)
+	if config == nil {
+		return nil, false, nil
+	}
+	return config.GetPlatformComponentConfig(), found && config.GetPlatformComponentConfig() != nil, err
+}
+
+func (d *datastoreImpl) UpsertPlatformComponentConfigRules(ctx context.Context, rules []*storage.PlatformComponentConfig_Rule) (*storage.PlatformComponentConfig, error) {
+	if ok, err := administrationSAC.WriteAllowed(ctx); err != nil {
+		return nil, err
+	} else if !ok {
+		return nil, sac.ErrResourceAccessDenied
+	}
+
+	config, found, err := d.store.Get(ctx)
```

No need to initialize the whole config here. You can just return an error if the entire config doesn't exist in the db.

---

I can't, because of this stack of calls:
- https://github.com/stackrox/stackrox/blob/86947b72195fe29a819bd4113d79fb9e93a686b0/central/config/datastore/singleton.go#L182
  - https://github.com/stackrox/stackrox/blob/86947b72195fe29a819bd4113d79fb9e93a686b0/central/config/datastore/datastore.go#L205
If this method returns an error, `UpsertConfig` returns an error, and then the `.must()` call in the config singleton initialization fails, and thus Central fails to start. I could modify it returning the nested error in `UpsertConfig`, but I feel like that would lead to potential issues down the line where it's silently failing to update the platform component config when trying to update the config via that method